index.js

```diff
@@ -16,11 +16,17 @@ const toObjectList = props => prop => ({key: prop, value: props[prop]})
 const normalizeClassName = prop => prop.key.toLowerCase() === 'classname' ? ({key: 'class', value: prop.value}) : prop
 const htmlForToFor = prop => prop.key === 'htmlFor' ? ({key: 'for', value: prop.value}) : prop
 
+const addEventToElement = (element, eventKey, eventValue) => {
+  element[eventKey] = eventValue
+  // add event to element list of events
+  element.events = element.events ? element.events.concat(eventKey) : [eventKey]
+}
+
 // handlers that can be filtered on
-// if something gets processed, we return false (using ![])
+// if something gets processed, we return false
 // otherwise, it returns true, indicating that this thing needs to be processed
-const handleOnAttrSetter = element => prop => prop.key.slice(0, 2) === 'on' ? ![element[prop.key] = prop.value] : true
-const handleNamespaceAttrSetter = (element, namespace) => prop => namespace ? ![element.setAttributeNS(null, prop.key, prop.value)] : true
+const handleOnAttrSetter = element => prop => prop.key.slice(0, 2) === 'on' ? addEventToElement(element, prop.key, prop.value) || false : true
+const handleNamespaceAttrSetter = (element, namespace) => prop => namespace ? element.setAttributeNS(null, prop.key, prop.value) || false : true
```

Comment:
isn't this the same as 
```js
namespace || !element.setAttributeNS(null, prop.key, prop.value)
```
also isn't `A || false` always going to be `A`? Or is this because it requires a boolean value even if undefined?

Reply:
Ahh, after reading the top a bit more I think I get some of the logic, I still feel like you should be watching how you are using ```A || false```